doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,281 @@
+* The SmartSwitch host PMON should be able to Power Cycle, Shutdown, Reset, and rest the PCIe link per DPU or the entire system
+* The DPU must provide additional information such as reboot cause, timestamp, etc as explained in the scheme once it boots its OS to DPU_STATE table.
+* When the DPU reboots itself, should log the reboot cause and update the previous-reboot-cause-dpu field in the ChassisStateDB when it boots up again
+* When the SmartSwitch host reboots the DPU, the host should update the previous-reboot-cause-host field in the ChassisStateDB
+* The reboot-cause history should provide a holistic view of the reboot cause of the SmartSwitch host CPU, the reboot-cause of all the DPUs as seen by the Switch and the DPU itself.
+* The DPUs should be uniquely identified (See DPU-ID IP/MAC allocation table) and the DPU upon boot should get this ID from the host and identify itself.
+* The DPU-ID itself can be used to assign the IP address of the midplane interface for both end points of the midplane interface between the host and dpu (see the midplane-interface  IP address assignment section)
+* Implement the required new APIs for DPU management (see details in design section)
+* SmartSwitch should use the existing SONiC midplane-interface model in modular chassis design for communication between the DPU and the NPU
+* SmartSwitch should extend the SONiC modular chassis design and treat the dpu-cards (SLED) just like line-cards in existing design
+* Reboot
+    * Only cold reboot of DPUs is required, warm boot support is not required.
+### 2.2. Monitoring and Thermal Management
+* Dpu State
+    * The DPUs should provide their state to the host as the boot progression happens, through the host hardware (see the DPU state table for details)
+    * SmartSwitch should store the dpu state data in the DPU_STATE table in the host ChassisStateDB (explained in DB schema)
+    * DPUs should be able to store the data using a redis call
+    * The DPU must provide additional information on the state once it boots its OS to DPU_STATE table.
+    * The SmartSwitch host PMON should be able to monitor the liveliness of the DPUs and when they go down should be able to take appropriate actions such as updating the state of the DPU in the DB and should try to gracefully recover the DPU when requested by the PMON
+
+* Thermal management
+    * Sensor values and fan speeds, status should be read periodically and stored in SmartSwitch StateDB
+    * Platform modules should use this thermal sensor values against the thresholds in the thermal policy and adjust fan speeds depending on the temperature
+    * Trigger thermal shut down on critical policy violation
+
+* Show CLIs
+    * Extend existing CLIs such as show platform fan/temperature to support the new HW
+    * Add new CLIs 
+        ```
+        show dpu <id> state
+        ```
+### 2.3. Detect and Debug
+* Health
+    * SmartSwitch DPUs should store their health data in their local StateDB 
```

Comment:
we should have a summary health state on chassis db

Reply:
A decision was made not to store this data on the chassisStateDB as this is high volume data.